membership-attribute-service/app/controllers/AccountController.scala

```diff
@@ -103,6 +100,73 @@ class AccountController(commonActions: CommonActions, override val controllerCom
     }
   }
 
+  private def updateDD[P <: SubscriptionPlan.AnyPlan : SubPlanReads] = BackendFromCookieAction.async { implicit request =>
+    // TODO - refactor to use the Zuora-only based lookup, like in AttributeController.pickAttributes - https://trello.com/c/RlESb8jG
+
+    val updateForm = Form { tuple(
+      "accountName" -> nonEmptyText,
+      "accountNumber" -> nonEmptyText,
+      "sortCode" -> nonEmptyText
+    ) }
+
+    val tp = request.touchpoint
+    val maybeUserId = authenticationService.userId
+    logger.info(s"Attempting to update direct debit for $maybeUserId")
+    (for {
+      user <- EitherT(Future.successful(maybeUserId \/> "no identity cookie for user"))
+      directDebitDetails <- EitherT(Future.successful(updateForm.bindFromRequest().value \/> "no direct debit details submitted with request"))
+      (bankAccountName, bankAccountNumber, bankSortCode) = directDebitDetails
+      sfUser <- EitherT(tp.contactRepo.get(user).map(_.flatMap(_ \/> s"no SF user $user")))
+      subscription <- EitherT(tp.subService.current[P](sfUser).map(_.headOption).map (_ \/> s"no current subscriptions for the sfUser $sfUser"))
+      account <- EitherT(annotateFailableFuture(tp.zuoraService.getAccount(subscription.accountId), s"get account with id ${subscription.accountId}"))
+      billToContact <- EitherT(annotateFailableFuture(tp.zuoraService.getContact(account.billToId), s"get billTo contact with id ${account.billToId}"))
+      bankTransferPaymentMethod = BankTransfer(
+        accountHolderName = bankAccountName,
+        accountNumber = bankAccountNumber,
+        sortCode = bankSortCode,
+        firstName = billToContact.firstName,
+        lastName = billToContact.lastName,
+        countryCode = "GB"
+      )
+      createPaymentMethod = CreatePaymentMethod(
+        accountId = subscription.accountId,
+        paymentMethod = bankTransferPaymentMethod,
+        paymentGateway = account.paymentGateway.get, // this will need to change to use this endpoint for 'payment method' SWITCH
+        billtoContact = billToContact,
+        invoiceTemplateOverride = None
+      )
+      _ <- EitherT(annotateFailableFuture(tp.zuoraService.createPaymentMethod(createPaymentMethod), "create direct debit payment method"))
+      freshDefaultPaymentMethodOption <- EitherT(annotateFailableFuture(tp.paymentService.getPaymentMethod(subscription.accountId), "get fresh default payment method"))
+    } yield freshDefaultPaymentMethodOption match {
+      case Some(dd: GoCardless)
+        if bankAccountName == dd.accountName &&
```

Comment:
We could pull this comparison into a def (or a val) to make it easier to follow this code

Reply:
since pulling the yield block into its own def (as per one of your other comments) pulling this if statement into its own thing seems like overkill - each clause of the if is on its own line and it being in a separate val/def would be no different

Reply:
happy to change if you still think it would help

Reply:
Cool, no need to change this on my account 👍